Modules/Filtering/ImageGrid/test/itkResampleImageTest2s.cxx

```diff
@@ -0,0 +1,267 @@
+
+  typedef itk::ImageFileReader< ImageType > ReaderType;
+  typedef itk::ImageFileWriter< ImageType > WriterType;
+
+  ReaderType::Pointer reader1 = ReaderType::New();
+  ReaderType::Pointer reader2 = ReaderType::New();
+  ReaderType::Pointer reader3 = ReaderType::New();
+  ReaderType::Pointer reader4 = ReaderType::New();
+
+  WriterType::Pointer writer1 = WriterType::New();
+  WriterType::Pointer writer2 = WriterType::New();
+  WriterType::Pointer writer3 = WriterType::New();
+  WriterType::Pointer writer4 = WriterType::New();
+
+  reader1->SetFileName( argv[1] );
+  reader2->SetFileName( argv[2] );
+  reader3->SetFileName( argv[3] );
+  reader4->SetFileName( argv[4] );
+
+  writer1->SetFileName( argv[3] );
+  writer2->SetFileName( argv[4] );
+  writer3->SetFileName( argv[5] );
+  writer4->SetFileName( argv[6] );
+
+  // Create an affine transformation
+  AffineTransformType::Pointer affineTransform = AffineTransformType::New();
+  affineTransform->Scale(2.0);
+
+  // Create a linear interpolation image function
+  InterpolatorType::Pointer interpolator = InterpolatorType::New();
+
+  // Create a nearest neighbor extrapolate image function
+  ExtrapolatorType::Pointer extrapolator = ExtrapolatorType::New();
+
+  // Create and configure a resampling filter
+  typedef itk::ResampleImageFilter< ImageType, ImageType > ResampleFilterType;
+
+  ResampleFilterType::Pointer resample = ResampleFilterType::New();
+
+  typedef itk::PipelineMonitorImageFilter< ImageType > MonitorFilter;
+  
+  MonitorFilter::Pointer monitor = MonitorFilter::New();
+
+  EXERCISE_BASIC_OBJECT_METHODS( resample, ResampleImageFilter, ImageToImageFilter );
+
+  resample->SetInput( reader1->GetOutput() );
+  TEST_SET_GET_VALUE( reader1->GetOutput(), resample->GetInput() );
+
+  resample->SetReferenceImage( reader2->GetOutput() );
+  TEST_SET_GET_VALUE( reader2->GetOutput(), resample->GetReferenceImage() );
+
+  resample->UseReferenceImageOn();
+  TEST_EXPECT_TRUE( resample->GetUseReferenceImage() );
+
+  resample->SetTransform( affineTransform );
+  TEST_SET_GET_VALUE( affineTransform, resample->GetTransform() );
+
+  resample->SetInterpolator( interpolator );
+  TEST_SET_GET_VALUE( interpolator, resample->GetInterpolator() );
+
+  monitor->SetInput( resample->GetOutput() );
+  writer1->SetInput( monitor->GetOutput() );
+
+  // Check GetReferenceImage
+  if( resample->GetReferenceImage() != reader2->GetOutput() )
+    {
+    std::cerr << "GetReferenceImage() failed ! " << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  // Run the resampling filter with the normal, linear, affine transform.
+  // This will use ResampleImageFilter::LinearThreadedGenerateData().
+  std::cout << "Test with normal AffineTransform." << std::endl;
+  try
+    {
+    writer1->SetNumberOfStreamDivisions(8); //split into 8 pieces for streaming.
+    writer1->Update();
+    }
+  catch( itk::ExceptionObject & excp )
+    {
+    std::cerr << excp << std::endl;
+    return EXIT_FAILURE;
+    }
```

Comment:
We can save typing and use the `TRY_EXPECT_NO_EXCEPTION` macro here.

Reply:
resolved in https://github.com/romangrothausmann/ITK/commit/0c738df46521afcaa0366a57e9f6a712db18a851